projects/demo/src/app/app.component.html

```diff
@@ -1,106 +1,74 @@
-        </val-errors>
-      </div>
+    <div class="flex-fill">
+      <h2>ngx&#8209;valdemort <small>by&nbsp;Ninja&nbsp;Squad</small></h2>
+      <p class="lead">
+        Simple, consistent validation error messages for your Angular forms.
+      </p>
     </div>
+  </div>
 
-    <button class="btn btn-primary">Submit</button>
+  <h2>The problem</h2>
 
-  </form>
+  <demo-problem></demo-problem>
 
-  <hr/>
+  <h2>Our solution: ngx-valdemort</h2>
 
-  <form ngSubmit="submitPerson()">
-    <div class="form-group">
-      <label for="name">Name (required)</label>
-      <input id="name" class="form-control" name="name" [(ngModel)]="person.name" required #n="ngModel"/>
-      <val-errors controlName="name" label="The name"></val-errors>
-    </div>
+  <demo-solution></demo-solution>
 
-    <div ngModelGroup="address">
-      <div class="form-group">
-        <label for="street">Street (required)</label>
-        <input id="street" class="form-control" name="street" [(ngModel)]="person.address.street" required />
-        <val-errors controlName="street" label="The street"></val-errors>
-      </div>
-
-      <div class="form-group">
-        <label for="city">City (required)</label>
-        <input id="city" class="form-control" name="city" [(ngModel)]="person.address.city" required #cityCtrl="ngModel" />
-        <val-errors [control]="cityCtrl.control" label="The city"></val-errors>
-      </div>
-    </div>
+  <h2>Consistent error messages</h2>
 
-    <button class="btn btn-primary">Submit</button>
-  </form>
+  <demo-consistency></demo-consistency>
 
-  <hr/>
+  <h2>Configuration</h2>
 
-  <div class="form-group">
-    <input [(ngModel)]="search" #s="ngModel" placeholder="Search" class="form-control" pattern="^[0-9]*$"/>
-    <val-errors [control]="s.control" label="The search"></val-errors>
-  </div>
+  <demo-configuration></demo-configuration>
 
-  <div class="form-group">
-    <input [formControl]="searchCtrl" placeholder="Search reactively" class="form-control" />
-    <val-errors [control]="searchCtrl" label="The reactive search"></val-errors>
-  </div>
+  <h2>Features</h2>
 
-  <hr/>
+  <ul class="features">
+    <li>Minimum markup needed</li>
+    <li>Consistent, default error messages</li>
+    <li>Generic or control-specific default error messages by specifying a label</li>
+    <li>Overriding of default messages for custom error messages</li>
+    <li>All error messages have access to the error value (min, max, required length, etc.)</li>
+    <li>I18n possible</li>
+    <li>HTML in messages possible</li>
+    <li>Use of pipes in messages possible</li>
+    <li>Usable in reactive forms, template-driven forms, or standalone controls</li>
+    <li>Usable on controls nested in form groups or form arrays</li>
+    <li>Central configuration service</li>
+    <li>Free, and open-source. <a href="https://github.com/Ninja-Squad/ngx-valdemort">The code is available on Github</a></li>
+  </ul>
```

Comment:
This list is a little bit arid to read. I have no great idea to improve, except maybe using images or icons for each features like we do for [the ebook](https://books.ninja-squad.com/angular).

Reply:
Yeah, that's my feeling, too. Maybe we should simply drop it, given that almost everything it says is explained in the above examples anyway.

Reply:
It can go as is for the first step, and we'll try to improve later. I think a few images/icons could help, and it's nice to have all the features on the page wihtout having to read everything.